### PR TITLE
Update minification and sourcemap settings on CI builds for sentry

### PR DESCRIPTION
**Pull request by novocaine**
With previous settings, our JS files for develop are so large that sentry's webserver rejects the upload.

In this change - 

* re-enable minification to reduce the size of the files
* update the CI sourcemap setting from `eval-source-map` to `source-map` to move the embedded source out of the .js payload and into .js.map files

This step fixed the issue for me locally at least by reducing the size of `init.js` from ~25MB to a 2MB js + 10MB .map

This PR also includes a careful refactor of the loading of build config depending on the args and environment, its best to follow this commit by commit.

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

### CODE REVIEW

**Comment by dbkr** [comment]
Looks sensible. I can't see any reason un-minified sources would be useful if we ship the app with source maps anyway, whether or not we publish them to sentry - would be good to double check from others.

**Comment by novocaine** [comment]
I realised that the problem is actually the embedded source is huge, so trying with `source-map` which splits the embedded source into a different `.map` file. Not really sure whether our deployment script will upload `.map` files - I think it just bindly unpacks the tar file?

File: webpack.config.js [hunk 1]
```diff
@@ -40,31 +40,34 @@ function getActiveThemes() {
...
+    //      (called to build prod, nightly and develop.element.io)
+    // arg.mode is set to "delopment" by yarn start
+    //      (called by developers, runs the continuous reload script)
+    // process.env.CI_PACKAGE is set when yarn build is called from scripts/ci_package.sh
+    //      (called to build nightly and develop.element.io)
+    const nodeEnv = argv.mode;
     const devMode = nodeEnv !== 'production';
     const useHMR = process.env.CSS_HOT_RELOAD === '1' && devMode;
     const fullPageErrors = process.env.FULL_PAGE_ERRORS === '1' && devMode;
+    const enableMinification = argv.mode === "production";
```
Comment (turt2live) [inline webpack.config.js:55]: I'd prefer if we instead used `!devMode` or otherwise relied upon `nodeEnv` as a variable

**Review by turt2live — APPROVED** [review]
otherwise this seems fine - developer machines won't explode, and seems to result in better bundles for end users. Assuming Sentry works with this, let's do it.

**Pull request merged**